src/server/game/Entities/Player/Player.cpp

```diff
@@ -27243,3 +27247,186 @@ uint32 Player::DoRandomRoll(uint32 minimum, uint32 maximum)
 
     return roll;
 }
+
+uint32 Player::PlayScene(uint32 sceneId, Position const* position /*= nullptr*/)
+{
+    SceneTemplate const* sceneTemplate = sObjectMgr->GetSceneTemplate(sceneId);
+    return PlaySceneByTemplate(sceneTemplate);
+}
+
+uint32 Player::PlaySceneByTemplate(SceneTemplate const* sceneTemplate, Position const* position/* = nullptr*/)
+{
+    if (sceneTemplate == nullptr)
+        return 0;
+
+    SceneScriptPackageEntry const* entry = sSceneScriptPackageStore.LookupEntry(sceneTemplate->scenePackageId);
+
+    if (entry == nullptr)
+        return 0;
+
+    // By default, take player position
+    if (position == nullptr)
+    {
+        Position loc = GetPosition();
+        position = &loc;
```

Comment:
@jackpoz the thing is, your comment wasn't in any way constructive. Shauren had already given a constructive answer showing the problem and the solution. So long as submitter understands the mistake, and if not reads up on pointers/references to gain that understanding. An improvement is made and it's all good.

I seem to remember TC being an educational project. For me, when i started I had no real experience in c++. But looking at the code and learning through doing IMO is a good way to do things. I know I've personally learned a lot from this project. I don't think it helps anyone to try and put people off learning in this way. Especially when the actual problem was already pointed out.

Yes, to work in open source projects, you need a thick skin. But, I don't think it helps anyone to turn people away from an educational project in this manner. PRs are a gating system to prevent such mistakes getting into core, and it's working as intended.